Feature/user email login

MauricioGamarraVargas opened the pull request.

MauricioGamarraV committed:
improve user login

-create a new stack auth
-create new component new user form signup and loginForm
-create new login action
-create new userEmailReducer

douglas-campos-coderoad-com commented on src/actions/LoginAction.js:
```diff
@@ -0,0 +1,50 @@
+        email: email,
+        createdAt: Date.now()
+      })
+      dispatch({ type: USER_SIGN_UP, payload: {firstName, lastName, signUpError: ''}})
```
please add semicolon at the end when is needed `;`

douglas-campos-coderoad-com commented on src/components/auth/ButtonSignUp.js:
```diff
@@ -0,0 +1,36 @@
+import LoginEmailForm from './LoginEmailForm';
+import { FontWeight, Size, Colors, Padding } from '../../constants/Styles';
+import SignUpEmailForm from './SignUpEmailForm';
+import { create } from 'lodash';
```
is needed this import?

douglas-campos-coderoad-com commented on src/components/auth/ButtonSignUp.js:
```diff
@@ -0,0 +1,36 @@
+import { useToast } from 'react-native-fast-toast'
+import {createUser} from '../../actions'
+import { Input, Icon } from 'react-native-elements'
+import { Text, View } from 'react-native';
```
make sure to remove the unnecessary imports

douglas-campos-coderoad-com commented on src/components/auth/LoginEmail.js:
```diff
@@ -0,0 +1,64 @@
+  
+  const showToast = () =>{
+    if(error){
+      toast.show("Usuario no encontrado crea una cuenta", {
```
Usuario no encontrado, por favor crea una cuenta.

MauricioGamarraV committed:
improve login finished

-improve styles in toast and message 
-set errorSingUp wirh error catch 
-delete signupButton component

MauricioGamarraV committed:
fix user info 

-set values of payload action

MauricioGamarraV committed:
fix bug google login

-add route to store list before singup

douglas-campos-coderoad-com committed:
fix google log in

MauricioGamarraV committed:
usrer data google fixed

- change method to verify user from currentUser of firebase to onAuthStateChanged()
- verify if user exist to make a order
